app/controllers/categories_controller.rb

```diff
@@ -394,6 +394,16 @@ def search
 
     Category.preload_user_fields!(guardian, categories)
 
+    # Prioritize categories that start with the term, then top-level
+    # categories, then subcategories
+    categories =
+      categories.to_a.sort_by do |category|
+        [
+          category.name.downcase.starts_with?(term) ? 0 : 1,
+          category.parent_category_id.blank? ? 0 : 1,
+        ]
+      end
```

Comment:
Just noticed that we implemented custom logic for categories search here. Is there a reason we don't use https://github.com/discourse/discourse/blob/main/lib/search.rb for search categories instead?

Reply:
Icic. I need to have a closer look at the code but I wonder if Search#category_search should be updated to use the same logic here.

Going to approve first as we can work on this in the future.